cli/internal/runcache/runcache.go

```diff
@@ -155,30 +125,54 @@ type TaskCache struct {
 	LogFileName       fs.AbsolutePath
 }
 
+// getTaskOutputMode returns the output mode for a given task with support for overriding it using the flag
+func (tc TaskCache) getTaskOutputMode() (util.TaskOutputMode, error) {
```

Comment:
Similarly, this feels like it should be a field set on `TaskCache`, rather than a method called. We have all .of the inputs when creating the `TaskCache` instance.